packages/client-config/src/paths/get_client_config_path.ts

```diff
@@ -23,11 +23,15 @@ export const getClientConfigPath = async (
   let targetPath = defaultArgs.out;
 
   if (outDir) {
-    const outDirIsFile = fs.lstatSync(outDir).isFile();
-    if (!outDirIsFile) {
-      targetPath = path.isAbsolute(outDir)
-        ? outDir
-        : path.resolve(process.cwd(), outDir);
+    targetPath = path.isAbsolute(outDir)
+      ? outDir
+      : path.resolve(process.cwd(), outDir);
+
+    try {
+      await fsp.access(outDir);
+    } catch (error) {
+      // outDir does not exist, so create dir
```

Comment:
we should assert that error indicates "dir not exists". There might be other error cases here that should bubble up.